resources/views/pages/message-thread/[messageThread].blade.php

```diff
@@ -1,53 +1,37 @@
 <?php
 
-use App\Enums\UserPreference;
+use App\Community\Services\MessageThreadService;
 use App\Models\MessageThread;
-use App\Models\User;
 use App\Support\Shortcode\Shortcode;
-use Illuminate\Support\Carbon;
-use function Laravel\Folio\{middleware, name};
+use Illuminate\Support\Facades\Auth;
+use Illuminate\View\View;
 
-middleware(['auth', 'can:view,messageThread']);
+use function Laravel\Folio\{middleware, name, render};
+
+middleware(['auth', 'can:view,messageThread']); // TODO add 'verified' middleware
```

Comment:
I can view my Messages list, but attempting to open any of them returns a 401.

Reply:
Strange, I am not able to repro this locally after trying numerous threads. However, `MessageThreadPolicy::view()` is just a stub, so I've removed `can:view,messageThread` from the middleware call on the page.

Reply:
Now I get `Laravel\Folio\InlineMetadataInterceptor::{closure}(): Argument #2 ($messageThread) must be of type App\Models\MessageThread, string given, called in /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php on line 36` trying to view individual messages

Reply:
This one is a bit perplexing. I did find a broken route that I just pushed a commit to fix for, but I'm not sure if it's related. 

![Screenshot 2024-05-16 at 6 09 00 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/ea41df01-870a-418c-8fcf-4cbeae352dcd)

That error message does seem to be complaining about the 2nd arg of the `render()` hook on the message thread page.

On the latest commit, if you're still seeing the error, can you try running:

```
sail artisan cache:clear
sail artisan view:clear
```

Reply:
Same error. Tried as a couple other users too.

Reply:
Still can't repro on my end no matter what I try 🙁. I've tried various users and whatnot at this point, no luck.

The error seems to be suggesting that a string is being given for `$messageThread` instead of an actual `MessageThread` model.

Latest commit has a couple more fixes to try:
* Pass in a whole `$message` entity to the route instead of the ID
* Rename the Blade file to point directly to the `id` field.

If this still doesn't work, the nuclear option might be to read `request()->getPathInfo()` and query for the thread itself. But I'm very puzzled why I'm unable to repro this issue. When I land on the page from an inbox, I very clearly see Folio kicking in and loading the model:

![Screenshot 2024-05-16 at 8 22 14 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/23357108-6b83-4b72-accd-77533e6e23fb)

Reply:
I backtracked through your commits, and the one that mattered was renaming the file.
`[messageThread].blade.php` generates the error.
`[MessageThread:id].blade.php` works.

For what it's worth, `[MessageThread:id].blade.php` generates a not-authorized error with the `'can:view,messageThread'` middleware clause. Not sure if that helps.

It seems like this is not a desirable change. What can I do to further identify the problem?

Reply:
I'm beginning to wonder if there could possibly be something OS-specific going on. My suspicion is driven by a line from the Folio documentation, which seems to suggest there is some OS-based logic in Folio itself:

---

![Screenshot 2024-05-17 at 5 15 03 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/ead48f7a-502c-43ee-bfa4-382e116e8aad)

---

`[MessageThread:id]` should be synonymous with `[messageThread]`. Using `[MessageThread:id]` isn't desirable due to inconsistency, but should be functionally equivalent with just `[messageThread]`.

In latest, based on a hunch, I've added `protected $table = 'message_threads';` to the `MessageThread` model. There's nothing wrong with that model inferring the name of its table... but I'm wondering if there's something in Folio that's looking for `$table`.

With this change applied in the latest commit, can you rename [MessageThread:id] back to [messageThread] and see if the error still happens? If not, we can probably just proceed with [MessageThread:id].

Reply:
The comment there is because ":" is not a valid character in Windows paths, which is yet another reason we shouldn't name the file `[MessageThread:id]`.

fwiw, `[messagethread].blade.php` generates a 404, so having the capital T is causing it to try to do _something_.

This also works, but shouldn't be necessary:
```
render(function (View $view, string $messageThread, MessageThreadService $pageService) {
    $user = Auth::user();
    $currentPage = (int) (request()->input('page.number') ?? 1);

    $messageThread = MessageThread::find($messageThread);

    return $view->with($pageService->buildForMessageThreadViewData($user, $messageThread, $currentPage));
});
```

Reply:
I'm particularly nervous about different behavior happening in different environments. The only way we see the same behavior on both of our machines is if I coerce Folio into pointing directly at the `id` field for the `MessageThread` model on this page.

To align with the doc and per your feedback regarding Windows paths, I've updated the filename to `[MessageThread-id].blade.php`. This still works great on my end - if it works on yours I think we're probably safe. I'm very puzzled why this problem is happening at all - I suspect it is a bug with Folio itself.

Reply:
I can confirm `[MessageThread-id].blade.php` also works on my VM.

[This](https://stackoverflow.com/questions/43654601/laravel-eloquent-route-model-binding-not-working-with-multi-word-table-name) and [this](https://stackoverflow.com/questions/72241894/laravel-implicit-binding-routing-with-more-than-two-word) suggest it's a casing thing with multi-word model names, but I couldn't make that work.